Sources/PrimerSDK/Classes/PCI/User Interface/Text Fields/PrimerCardNumberFieldView.swift

```diff
@@ -87,6 +87,23 @@ public final class PrimerCardNumberFieldView: PrimerTextFieldView {
         }
 
         primerTextField.text = newText.withoutWhiteSpace.separate(every: 4, with: " ")
+
+        // Detect pasting of the card number
+        if string.count > 1 && (primerTextField.internalText?.count ?? 0) > 1 {
+            if let text = primerTextField.internalText {
+
+                // Get the position of the last character in the string
+                if let endPosition = primerTextField.position(from: primerTextField.beginningOfDocument, offset: text.count) {
+                    // Delaying it for 0.01 seconds because setting the selectedTextRange
+                    // before `shouldChangeCharactersIn:range` returns does not apply the value.
+                    DispatchQueue.main.asyncAfter(deadline: .now() + 0.01, execute: {
+                        // Create a UITextRange from the endPosition to endPosition (for placing the cursor at the end)
+                        primerTextField.selectedTextRange = primerTextField.textRange(from: endPosition, to: endPosition)
```

Comment:
If the user pastes into the middle of the string what happens?

It's an edge case but I think having it move the cursor to the end of the text doesn't feel right. Could we move the cursor to the end of the inserted range instead?